wp-includes/rest-api/endpoints/class-wp-rest-customize-changesets-controller.php

```diff
@@ -94,4 +133,835 @@ public function register_routes() {
+				),
+				'slug'            => array(
+					'description' => __( 'Unique Customize Changeset identifier, uuid' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'embed' ),
+					'arg_options' => array(
+						'sanitize_callback' => array( $this, 'sanitize_slug' ),
+					),
+					'readonly'   => true,
+				),
+				'status'          => array(
+					'description' => __( 'A named status for the object.' ),
+					'type'        => 'string',
+					'enum'        => $this->statuses,
+					'context'     => array( 'view', 'edit' ),
+					'arg_options' => array(
+						'sanitize_callback' => array( $this, 'sanitize_post_statuses' ),
+					),
+				),
+				'title'           => array(
+					'description' => __( 'The title for the object.' ),
+					'type'        => 'object',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'arg_options' => array(
+						'sanitize_callback' => null,
+					),
+					'properties'  => array(
+						'raw' => array(
+							'description' => __( 'Title for the object, as it exists in the database.' ),
+							'type'        => 'string',
+							'context'     => array( 'edit' ),
+						),
+						'rendered' => array(
+							'description' => __( 'HTML title for the object, transformed for display.' ),
+							'type'        => 'string',
+							'context'     => array( 'view', 'edit', 'embed' ),
+							'readonly'    => true,
+						),
+					),
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Checks if a given request has access to read a changeset.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return bool|WP_Error True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+
+		$post_type_obj = get_post_type_object( $this->post_type );
+		$changeset_post = $this->get_customize_changeset_post( $request['uuid'] );
+		if ( ! $changeset_post ) {
+			return false;
+		}
+		$data = array();
+		if ( isset( $request['customize_changeset_data'] ) ) {
+			$data = $request['customize_changeset_data'];
+		}
+
+		if ( 'edit' === $request['context'] && $changeset_post && ! $this->check_update_permission( $changeset_post, $data ) ) {
+			return new WP_Error( 'rest_forbidden_context', __( 'Sorry, you are not allowed to edit this post.' ), array(
+				'status' => rest_authorization_required_code(),
+			) );
+		}
+
+		return $this->check_read_permission( $post_type_obj, $changeset_post );
+	}
+
+	/**
+	 * Check if current user can read the changeset.
+	 *
+	 * @param object $post_type_obj Post type object.
+	 * @param object $changeset_post Changeset post object.
+	 * @return bool If has read permissions.
+	 */
+	protected function check_read_permission( $post_type_obj, $changeset_post ) {
+		return current_user_can( $post_type_obj->cap->read_post, $changeset_post->ID );
```

Comment:
So far I understood that if some of the settings are forbidden then those settings would be hidden from the response but other's could still be viewed. And that's why initially in `check_read_permission` only the `read` permission for the post would matter. Or would that make it confusing? Thoughts? Maybe it is confusing. For example in case of the user not having permissions to any of the settings.

This here was taking care of not showing the forbidden settings but showing the others:
https://github.com/WP-API/wp-api-customize-endpoints/blob/d0b5ceec631719d5bc865e7a341b46a87f949a8a/wp-includes/rest-api/endpoints/class-wp-rest-customize-changesets-controller.php#L403-L405